### ISSUE TITLE
Using default_pad_value if image is of type LABEL

### ISSUE THREAD

**Issue by bcrobo** [TorchIO-project/torchio#1304]
### Is there an existing issue for this?

- [x] I have searched the existing issues

### Bug summary

I'm using a `LabelMap` as part of my subject and my data augmentation relies on the `RandomAffine` class. I provided a `default_pad_value=X` where `X` is plain old integer and I was surprised to see that it was replaced by `0` in the augmented image. Indeed the code of random affine (see [here](https://github.com/TorchIO-project/torchio/blob/main/src/torchio/transforms/augmentation/spatial/random_affine.py#L396)) forces the value to be 0 when the image type is not `INTENSITY`. 

### Code for reproduction

```python
import numpy as np
import torch
import torchio as tio

label_data = torch.from_numpy(
  np.full((1, 2, 2, 2), 1)
)
s = tio.Subject(
  label=tio.LabelMap(tensor=label_data)
)
aff = tio.RandomAffine(
  p=1,
  translation=(-10, 10, -10, 10, -10, 10),
  default_pad_value=250
)
s_aug = aff.apply_transform(s)
s_aug['label'].tensor
```

### Actual outcome

tensor([[[[0., 0.],
          [0., 0.]],
         [[0., 0.],
          [0., 0.]]]])

### Error messages

```python-traceback

```

### Expected outcome

I would have expect the `default_pad_value` to be used here. Is there any reason why this is not possible ?

### System info

```shell
Platform:   Linux-6.8.0-59-generic-x86_64-with-glibc2.35
TorchIO:    0.20.8
PyTorch:    2.6.0+cu124
SimpleITK:  2.5.0 (ITK 5.4)
NumPy:      2.0.2
Python:     3.9.22 | packaged by conda-forge | (main, Apr 14 2025, 23:35:59)
[GCC 13.3.0]
```

**Comment by fepegar** [comment]
Thanks for reporting, @bcrobo. Maybe we can add a `default_pad_label` for label maps, which also defaults to `0`.

**Comment by fepegar** [comment]
New parameter available in `v0.20.15`.

**Comment by bcrobo** [comment]
@fepegar thank you for having the time to look at it.

Looking at the MR  I would have reused the `get_default_pad_value` function instead. I think we know from the `image[TYPE]` if this is a `tio.LabelMap` or not hence avoiding the `default_label_pad_value` extra-argument. It would have avoid to copy-paste the `get_default_x_value` code and reduce the code base... But there might be a limitation of the code I'm not aware of.

Still, I'm pretty happy to hear there's gonna be a fix in the next release.
Have a good day.

**Comment by romainVala** [comment]
just like we did for interpolation (we have `image_interpolation` and `label_interpolation`)
so we need 2 keyword `default_image_pad_value` and `default_label_pad_value`

**Comment by bcrobo** [comment]
Yeah my apologies. Sounds great. Thanks for the fix !
